### ISSUE TITLE
4.10 fails to build on Mac with Python 3.8

### ISSUE THREAD

**Issue by mauritsvanrees** [zopefoundation/zope.container#48]
TLDR: the master branch seems okay, so let's release it.

This is hard to describe, because there are so many variants and combinations of Python, pip, buildout you can try.
I see a problem in the Plone 5.2 core development buildout on Python 3.8, which uses latest zope.container 4.10. Likely Plone 6 will have the same problem. It has a few errors like this and then gives up:

```
src/zope/container/_zope_container_contained.c:115:11:
error: implicit declaration of function 'PyString_AS_STRING' is invalid in C99
[-Werror,-Wimplicit-function-declaration]
```

- With Python 3.8.16 and Buildout 2 (including the `wheel` package) it fails like above. It downloads the source distribution.
- With Python 3.8.16 and Buildout 3 it works. I get this egg: `zope.container-4.10-py3.8-macosx-12.6-x86_64.egg`. This is compiled from the wheel that it downloads: `zope.container-4.10-cp38-cp38-macosx_10_9_x86_64.whl`.
- When I try Buildout 2 again with the egg in the cache, it works.
- With Python 3.8.13 it always works. I get this egg: `zope.container-4.10-py3.8-macosx-10.15-x86_64.egg`. It compiles this from the wheel that it downloads: `zope.container-4.10-cp38-cp38-macosx_10_9_x86_64.whl`.

I am on MacOS Monterey 12.6.2. Starting from Python 3.8.14, this wants to have eggs/wheels for 12.6 and no longer accepts the 10.15 eggs. BTW, I made sure my eggs cache was empty before I started.

When I checkout `zope.container` with git and use the 4.10 tag, `bin/python setup.py bdist_wheel` always fails in the C code. Tried on Python 3.8 and 3.11.  The master branch is fine, because the C code has been fixed.

What it boils down to:

1. Depending on your exact versions of Python, pip, zc.buildout, setuptools, and wheel, Buildout may see a wheel on PyPI that it can use as basis. With recent versions, all is well.
2. If no wheel is found, the source distribution is used, and that always fails on Python 3.8.14+.
3. With a new 5.0 release, it looks like all will be well again.

So possible improvements:

1. A 5.0 release would help.
2. If more Mac wheels could be built and uploaded, this would avoid having to compile the C code from source. But this may be a never ending story, where we end up with 100 wheels to please everyone...

**Comment by icemac** [comment]
@mauritsvanrees Is this still an issue or were you able to solve it?

**Comment by mauritsvanrees** [comment]
It works with `zope.container` 5.0.
